src/lasuite/malware_detection/backends/jcop.py

```diff
@@ -0,0 +1,222 @@
+            raise FileNotFoundError(f"File {file_path} not found")
+
+        return analyse_file_async.delay(file_path, **kwargs)
+
+
+@shared_task(
+    bind=True,
+    default_retry_delay=30,
+    max_retries=3,
+    dont_autoretry_for=(MalwareDetectionInvalidAuthenticationError,),
+)
+def analyse_file_async(self, file_path: str, file_hash: str | None = None, **kwargs) -> None:
+    """Task starting analysis process for a file."""
+    if not file_hash:
+        file = default_storage.open(file_path, "rb")
+        file_hash = hashlib.file_digest(file, "sha256").hexdigest()
+        file.close()
+
+    callback = import_string(getattr(settings, setting_name("MALWARE_DETECTION_CALLBACK")))
+
+    api_key = getattr(settings, setting_name("JCOP_API_KEY"))
+    base_url = getattr(settings, setting_name("JCOP_BASE_URL"))
+    result_timeout = getattr(settings, setting_name("JCOP_RESULT_TIMEOUT"), 30)
+
+    try:
+        # try if the file as already been tested
+        response = requests.get(
+            f"{base_url}/results/{file_hash}",
+            headers={
+                "X-Auth-Token": api_key,
+                "Accept": "application/json",
+            },
+            timeout=result_timeout,
+        )
+    except requests.exceptions.RequestException as exc:
+        logger.error("Error getting cache result for file %s: %s", file_path, exc)
+        if self.request.retries >= self.max_retries:
+            callback(
+                file_path,
+                ReportStatus.UNKNOWN,
+                error_info={
+                    "error": "Max retries triggering new analysis exceeded",
+                    "error_code": 5000,
+                }
+                ** kwargs,
+            )
+        raise self.retry(exc=exc) from exc
+
+    if response.status_code == HTTPStatus.NOT_FOUND:
+        # start a new analysis
+        trigger_new_analysis.delay(file_path, **kwargs)
+        return
+
+    if response.status_code == HTTPStatus.UNAUTHORIZED:
+        callback(
+            file_path,
+            ReportStatus.UNSAFE,
+            error_info={
+                "error": "Invalid API key",
+                "error_code": response.status_code,
+            },
+            **kwargs,
+        )
+        raise MalwareDetectionInvalidAuthenticationError()
+
+    if response.status_code == HTTPStatus.OK:
+        content = response.json()
+        if not content["done"]:
+            # the analysis is not done yet, retry later
+            self.retry()
+            return
+
+        if content.get("error_code"):
+            status = ReportStatus.UNSAFE if content["is_malware"] else ReportStatus.UNKNOWN
+            callback(
+                file_path,
+                status,
+                error_info={
+                    "error": content["error"],
+                    "error_code": content["error_code"],
+                },
+                **kwargs,
+            )
+            return
+
+        if not content.get("is_malware"):
```

Comment:
Do we expect the key to be missing? 

Just for clarity sake, I would write
```suggestion
        if not content.get("is_malware", False):
```